sope-ldap/NGLdap/NGLdapConnection.m

```diff
@@ -194,10 +195,13 @@ - (NSException *)_exceptionForErrorCode:(int)_err
                            @"operation %@ failed: %@ (0x%X)",
                            _operation, ldapError, _err];
 
+  userInfo = [NSMutableDictionary dictionaryWithCapacity: 2];
```

Comment:
Here I have a problem with dictionaries usage, I think that instead of creating a mutable dictionary it would be better to extract the list of the contents of the dictionary, add the new object and use this list to create a NSDictionary. However I do not find a way to do this that is not more contrived that the NSMutableDictionary approach, can you suggest a better way to do this?